dot/config_test.go

```diff
@@ -0,0 +1,496 @@
+						"childstate", "syncstate", "payment"},
+					WSPort: 8546,
+				},
+				Pprof: PprofConfig{
+					Settings: pprof.Settings{
+						ListeningAddress: "localhost:6060",
+					},
+				},
+			},
+			configMaker: KusamaConfig,
+		},
+		{
+			name: "polkadot default",
+			want: &Config{
+				Global: GlobalConfig{
+					Name:         "Polkadot",
+					ID:           "polkadot",
+					BasePath:     "~/.gossamer/polkadot",
+					LogLvl:       log.Info,
+					MetricsPort:  9876,
+					RetainBlocks: 512,
+					Pruning:      "archive",
+				},
+				Log: LogConfig{
+					CoreLvl:           log.Info,
+					DigestLvl:         log.Info,
+					SyncLvl:           log.Info,
+					NetworkLvl:        log.Info,
+					RPCLvl:            log.Info,
+					StateLvl:          log.Info,
+					RuntimeLvl:        log.Info,
+					BlockProducerLvl:  log.Info,
+					FinalityGadgetLvl: log.Info,
+				},
+				Init: InitConfig{Genesis: "./chain/polkadot/genesis.json"},
+				Core: CoreConfig{
+					Roles:           byte(1),
+					WasmInterpreter: "wasmer",
+				},
+				Network: NetworkConfig{
+					Port: 7001,
+				},
+				RPC: RPCConfig{
+					Port: 8545,
+					Host: "localhost",
+					Modules: []string{"system", "author", "chain", "state", "rpc", "grandpa", "offchain",
+						"childstate", "syncstate", "payment"},
+					WSPort: 8546,
+				},
+				Pprof: PprofConfig{
+					Settings: pprof.Settings{
+						ListeningAddress: "localhost:6060",
+					},
+				},
+			},
+			configMaker: PolkadotConfig,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			got := tt.configMaker()
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func TestRPCConfig_isRPCEnabled(t *testing.T) {
+	t.Parallel()
+
+	type fields struct {
+		Enabled        bool
+		External       bool
+		Unsafe         bool
+		UnsafeExternal bool
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   bool
+	}{
+		{
+			name: "default",
+			want: false,
+		},
+		{
+			name:   "enabled true",
+			fields: fields{Enabled: true},
+			want:   true,
+		},
+		{
+			name:   "external true",
+			fields: fields{External: true},
+			want:   true,
+		},
+		{
+			name:   "unsafe true",
+			fields: fields{Unsafe: true},
+			want:   true,
+		},
+		{
+			name:   "unsafe external true",
+			fields: fields{UnsafeExternal: true},
+			want:   true,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			r := &RPCConfig{
+				Enabled:        tt.fields.Enabled,
+				External:       tt.fields.External,
+				Unsafe:         tt.fields.Unsafe,
+				UnsafeExternal: tt.fields.UnsafeExternal,
+			}
+			got := r.isRPCEnabled()
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func TestRPCConfig_isWSEnabled(t *testing.T) {
+	t.Parallel()
+
+	type fields struct {
+		WS               bool
+		WSExternal       bool
+		WSUnsafe         bool
+		WSUnsafeExternal bool
+	}
```

Comment:
remove `fields` and use `RPCConfig` directly

Reply:
updated.